File: cs50/pset2/vigenere/vigenere.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>

const int lower_low = 'a';
const int lower_high = 'z';
const int upper_low = 'A';
const int upper_high = 'Z';

//Check if all input is either between a-z or A-Z
//Start encrypting

int check_input(string input)
{
    for (int i = 0,n = strlen(input); i < n; i++)
    {
        if ((input[i] >= lower_low && input[i] <= lower_high) || (input[i] >= upper_low && input[i] <= upper_high))
        {
            continue;
        }
        else
        {
            return false;
        }
    }
    return true;
}

int main(int argc, string argv[])
{
    if (argc != 2)
    {
        printf("Usage: ./vigenere k\n");
        return 1;
    }
    else if (check_input(argv[1]) != true)
    {
        printf("Usage: ./vigenere k\n");
        return 1;
    }
    else
    {
        string input = get_string("plaintext:  ");
        printf("ciphertext: ");
        int counter = 0;

        for (int i = 0, n = strlen(input); i < n; i++)
        {
            int current = counter%strlen(argv[1]);



            if (input[i] >= lower_low && input[i] <= lower_high)
            {
                int lower = tolower(argv[1][current]);
                char result = (((input[i]+(lower-'a')-'a')%26)+'a');
                printf("%c", result);
                counter++;
            }
            else if (input[i] >= upper_low && input[i] <= upper_high)
            {
                int lower = tolower(argv[1][current]);
                char result = (((input[i]+(lower-'a')-'A')%26)+'A');
                printf("%c", result);
                counter++;
            }
            else
            {
                printf("%c", input[i]);
            }
        }
        printf("\n");
    }
}

File: cs50/pset6/sentimental/vigenere/vigenere/vigenere.py
import sys

lower_low = ord('a')
lower_high = ord('z')
upper_low = ord('A')
upper_high = ord('Z')


def validate_argv():
    if(len(sys.argv) != 2 or not sys.argv[1].isalpha() or (' ' in sys.argv[1]) == True):
        print("Usage: python vigenere.py k")
        return False
    else:
        return True


def main():
    output = ""
    if(not validate_argv()):
        exit(1)
    else:
        counter = 0
        string = input("plaintext: ")
        print("ciphertext: ", end="")
        for c in string:
            if (sys.argv[1][counter % len(sys.argv[1])].islower()):
                current = ord(sys.argv[1][counter % len(sys.argv[1])]) - lower_low
            else:
                current = ord(sys.argv[1][counter % len(sys.argv[1])]) - upper_low

            c2 = ord(c)
            if(c2 >= lower_low and c2 <= lower_high):
                print(chr((((c2 + current) - lower_low) % 26) + lower_low), end="")
                counter += 1
            elif(c2 >= upper_low and c2 <= upper_high):
                print(chr((((c2 + current) - upper_low) % 26) + upper_low), end="")
                counter += 1
            else:
                print(c, end="")

    print()
    return 0


main()
